pkg/controller/controller.go

```diff
@@ -437,32 +447,42 @@ func (c *controller) fetchRouteV1Alpha1(ctx context.Context, instance *v1alpha1.
 	fetchObiData := func() {
 		tmpFirstCall := firstCall
 		firstCall = false
+		// Get the runtime resource for 1st time if the OBI is based on label match
 		if tmpFirstCall || (targetRef.Name == "" && len(targetRef.Labels) > 0) {
-			unstructuredResource, err = c.getRuntimeObject(gvr, instance)
-			if err != nil {
-				return
+			var err error
+			if !gvr.Empty() {
+				unstructuredResource, err = c.getRuntimeObject(gvr, instance)
```

Comment:
Should get the upstream code, the definition of this `getRuntimeObject` function has changed.